pydra/engine/core.py

```diff
@@ -480,8 +482,8 @@ def _modify_inputs(self):
         modified_inputs = template_update(
             self.inputs, self.output_dir, map_copyfiles=map_copyfiles
         )
-        if modified_inputs:
-            self.inputs = attr.evolve(self.inputs, **modified_inputs)
+        for name, value in modified_inputs.items():
+            setattr(self.inputs, name, value)
```

Comment:
This is okay for now, but it feels messy. I think it would be good to find some way to keep using attrs in the usual way, and keep track of the changes inside `Task` instead of `Task.inputs._hashes`.

Reply:
In the syntax changes I am proposing, the input spec would be subsumed into the task class in any case. So I might hold off and see what people think of them first

Reply:
and why this is changing?  is this  related to the issue with pydra-ml?

Reply:
I can't actually remember why that was changed to be like that, but in any it has been changed further now with the changes to fix the pydra-ml-raised issue